assets/js/CheckboxForSelectImages.js

```diff
@@ -0,0 +1,142 @@
+// Code to be executed if favorite picture is present
+document.querySelector(".favorite-picture") && document.addEventListener("DOMContentLoaded", function () {
+
+    console.log("favorite-picture is present");
+    // Get the current URL
+    var currentUrl = window.location.href;
+    var currentEntity = null;
+
+    // Check if the URL contains "album" or "edit"
+    if (currentUrl.includes("album")) {
+        console.log("The URL contains the word 'album'");
+        currentEntity = "album";
+    } else if (currentUrl.includes("category")) {
+        console.log("The URL contains the word 'category'");
+        currentEntity = "category";
+    } else {
+        console.log("Neither 'album' nor 'category' found in the URL");
+        return;
+    }
+
+
+
+    const checkboxes = document.querySelectorAll(".favorite-picture");
+
+    checkboxes.forEach(function (checkbox) {
+        checkbox.addEventListener("change", function () {
+            parent_id = checkbox.dataset.parentId;
+            photo_id = checkbox.dataset.photoId;
+
+            checkedCheckboxes = Array.from(checkboxes).filter(function (cb) {
+                return cb.checked;
+            });
+            checkedCheckboxes.forEach(function (otherCheckbox) {
+                if (otherCheckbox !== checkbox) {
+                    otherCheckbox.checked = false;
+                    otherCheckbox.parentNode.classList.remove("favorite");
+                }
+            });
+            // Determine the value to send in the AJAX request
+            const dataToSend = {
+                photoId: this.checked ? photo_id : null, // Send photo_id if checked, otherwise send null
+            };
+
+            // Send an AJAX request to update favorite picture
+            $.ajax({
+                url: "/admin/" + currentEntity + "/" + parent_id + "/update/favorite", // Replace with your actual route
+                type: "POST",
+                data: dataToSend,
+                success: function (response) {
+                    checkbox.parentNode.classList.add("favorite");
+                },
+                error: function (response) {
+                    alert("Erreur! " + response.responseText);
+                },
+            });
+        });
+    });
+});
+
+
+// Code to be executed if favorite picture is present
+document.querySelector(".photoToDelete") && document.addEventListener("DOMContentLoaded", function () {
+    console.log("PhotoToDelete is present");
```

Comment:
this line too

Reply:
Done, removed all unnecessary console logs